arangod/GeneralServer/AcceptorTcp.cpp

```diff
@@ -165,8 +169,12 @@ void AcceptorTcp<SocketType::Tcp>::asyncAccept() {
     info.encryptionType = _endpoint->encryption();
     info.serverAddress = _endpoint->host();
     info.serverPort = _endpoint->port();
-
-    std::unique_ptr<AsioSocket<SocketType::Tcp>> as = std::move(_asioSocket);
+    
+    std::unique_ptr<AsioSocket<SocketType::Tcp>> as;
+    {
+      std::unique_lock<std::mutex> guard(_mutex);
+      as = std::move(_asioSocket);
```

Comment:
I wonder if it is possible that another thread can call `close()` between the beginning of this callback and this line. If so, then the `close()` could reset the `_asioSocket` pointer, which we then use here and dereference.
Wouldn't it be better to lock the mutex at the beginning of the callback, move out `_asioSocket` under the lock, and then release it?
We would also need an extra check that the moved-out `_asioSocket` is not a nullptr, of course.